packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -2290,6 +2292,15 @@ class _HighlightModeManager {
     return handled;
   }
 
+  void handleSemanticsAction(SemanticsActionEvent semanticsActionEvent) {
+    if (kIsWeb &&
+        semanticsActionEvent.type == SemanticsAction.focus &&
+        _lastInteractionWasTouch != true) {
+      _lastInteractionWasTouch = true;
```

Comment:
It is counter-intuitive that code that handles `SemanticsAction.focus` sets `_lastInteractionWasTouch` to true here. Perhaps it's time we renamed `_lastInteractionWasTouch` to something that reflects the intent, e.g. `_lastInteractionRequiresNoHighlight` (or if we want to remove double-negatives `_lastInteractionRequiresHighlight`).

Reply:
Renaming this is a good idea.